### PR TITLE
Ecma404 support

### PR DESCRIPTION
**Pull request by schlunz**
Dear @otoolep,

after redesigning the parser, I was able to implement the ecma404 support.

### CODE REVIEW

**Comment by otoolep** [comment]
Let me know when the tests pass and I'll take a look.

File: input/parser_test.go [hunk 1]
```diff
@@ -252,7 +252,18 @@ func Test_Parsing(t *testing.T) {
 			if !ok {
 				t.Error("\n\nParser should succeed.\n")
 			}
-			if !reflect.DeepEqual(p.Result, tt.expected) {
+			ckeckEquality := func() bool {
+				for k, v := range p.Result {
```
Comment (schlunz) [inline input/parser_test.go]: For some reason, I this equality test (as well as `reflect.DeepEqual()` does not work.

**Comment by schlunz** [comment]
@otoolep For some reason I can't get the parser_tets running (see comment). Do you have any idea what I implemented wrong?

**Comment by otoolep** [comment]
Not sure, I'll have to leave it with you for now.

**Pull request closed without merging**